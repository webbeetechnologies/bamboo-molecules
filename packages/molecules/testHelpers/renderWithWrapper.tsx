import type { ReactElement, JSXElementConstructor, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ProvideMolecules } from '../src';

export const renderWithWrapper = <P extends {} = {}>(
    component: ReactElement<P, string | JSXElementConstructor<P>>,
    options?: RenderOptions,
): ReturnType<typeof render> => {
    const Wrapper = ({ children }: { children: ReactNode }) => (
        <ProvideMolecules>{children}</ProvideMolecules>
    );

    return render(component, { wrapper: Wrapper, ...options });
};
