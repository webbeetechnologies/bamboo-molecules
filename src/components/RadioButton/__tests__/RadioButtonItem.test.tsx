import { renderWithWrapper } from '../../../../testHelpers';
import { RadioButton } from '../index';

it('renders unchecked', () => {
    const tree = renderWithWrapper(
        <RadioButton.Item status="unchecked" label="Unchecked Button" value="unchecked" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

it('can render leading radio button control', () => {
    const tree = renderWithWrapper(
        <RadioButton.Item
            label="Default with leading control"
            status={'unchecked'}
            value="default"
            position="leading"
        />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
