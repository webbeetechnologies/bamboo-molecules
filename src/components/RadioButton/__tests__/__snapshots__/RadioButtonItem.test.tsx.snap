// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render leading radio button control 1`] = `
<View
  accessibilityLabel="Default with leading control"
  accessibilityRole="radio"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": false,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      false,
      Object {},
    ]
  }
>
  <View
    pointerEvents="none"
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <View
      accessibilityLiveRegion="polite"
      accessibilityRole="radio"
      accessibilityState={
        Object {
          "checked": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 18,
            "padding": 6,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <Text
          allowFontScaling={false}
          color="rgba(73, 69, 79, 1)"
          name="check"
          size={24}
          style={
            Object {
              "animationScale": 1,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "color": undefined,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontSize": 16,
          "textAlign": "right",
        }
      }
    >
      Default with leading control
    </Text>
  </View>
</View>
`;

exports[`renders unchecked 1`] = `
<View
  accessibilityLabel="Unchecked Button"
  accessibilityRole="radio"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": false,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      false,
      Object {},
    ]
  }
>
  <View
    pointerEvents="none"
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": undefined,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontSize": 16,
          "textAlign": "left",
        }
      }
    >
      Unchecked Button
    </Text>
    <View
      accessibilityLiveRegion="polite"
      accessibilityRole="radio"
      accessibilityState={
        Object {
          "checked": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 18,
            "padding": 6,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <Text
          allowFontScaling={false}
          color="rgba(73, 69, 79, 1)"
          name="check"
          size={24}
          style={
            Object {
              "animationScale": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
